infra/utils.sh

```diff
@@ -0,0 +1,173 @@
+        printf "allow-sa-permissions(project, account) requires 1 or 2 arguments" >&2
+        return 1
+    fi
+    local project="$1"
+    local account="$2"
+    local full_name_account=$(svc_acct_for "${project}" "${account}")
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/logging.logWriter
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/monitoring.viewer
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/monitoring.metricWriter
+}
+
+# Ensure the program is installed
+function check-install() {
+    if ! command -v "${1}" &>/dev/null; then
+        color 4 "You don't have ${1} installed. Please install ${1}."
+        exit 1
+    fi
+}
+
+# Create a BiqQuery Dataset for usage metering
+# $1 : The GCP project name
+# $2 : The BigQuery dataset name
+# $3 : The dataset region (optional)
+function create-bigquery-dataset() {
+    if [ $# -lt 1 ] && [ $# -gt 2 ] && [ -z "$1" ]; then
+        printf "create-bigquery-dataset(project, dataset_name) requires 1 or 2 arguments" >&2
+        return 1
+    fi
+    local -r project="$1"
+    local -r dataset_name="$2"
+    local -r region="${US:-}"
+    if ! bq --project "${project}" ls "${dataset_name}" >/dev/null 2>&1; then
+        bq mk --project_id="${project}" --location="${region}" "${dataset_name}"
+    else
+        printf "Found existing %s dataset.\n" "${dataset_name}"
+    fi
+}
+
+function create-cluster() {
+    gcloud beta container clusters create "${NAME}" \
+        --project "${PROJECT}" \
+        --region "${REGION}" \
+        --machine-type "${MACHINE_TYPE}" \
+        --disk-type "${DISK_TYPE}" \
+        --cluster-version "${CLUSTER_VERSION}" \
+        --num-nodes "${NUM_NODES}" \
+        --enable-network-policy \
+        --scopes "${SCOPES}" \
+        --network "${NAME}" \
+        --subnetwork "${NAME}-custom-subnet" \
+        --enable-ip-alias \
+        --enable-private-nodes \
+        --enable-master-authorized-networks \
+        --master-ipv4-cidr "${MASTER_IPV4_CIDR}" \
+        --master-authorized-networks "${MASTER_IP_RANGE}" \
+        --max-pods-per-node "${MAX_PODS_PER_NODE}" \
+        --no-issue-client-certificate \
+        --metadata disable-legacy-endpoints=true \
+        --service-account "$(svc_acct_for "${PROJECT}" "${SVC_ACCOUNT}")" \
+        --security-group "gke-security-groups@kubernetes.io" \
+        --enable-stackdriver-kubernetes \
+        --resource-usage-bigquery-dataset "${DATASET_NAME}" \
+        --quiet
+
+    if ! gcloud beta container clusters describe test-cluster --format="value(resourceUsageExportConfig)" >/dev/null 2>&1; then
+        gcloud beta container clusters update "${NAME}" --region "${REGION}" --resource-usage-bigquery-dataset "${DATASET_NAME}"
```

Comment:
I don't grok what this is doing?

Reply:
Verify that usage metering is enabled for the cluster.  I'll add a comment